adapter/src/hermes/adapter/stdio/stdio.cc

```diff
@@ -455,17 +536,33 @@ int HERMES_DECL(fclose)(FILE *fp) {
           hermes::api::VBucket file_vbucket(filename, mdm->GetHermes(), true,
                                             ctx);
           auto offset_map = std::unordered_map<std::string, hermes::u64>();
+          std::size_t pos = 0;
           for (const auto &blob_name : blob_names) {
             file_vbucket.Link(blob_name, filename, ctx);
-            offset_map.emplace(blob_name, std::stol(blob_name) * kPageSize);
+            /* FIXME(hari): change this once we have blob namespace separated
+             * per bucket.*/
+            if (pos == 0) {
+              pos = blob_name.find(kStringDelimiter) + 1;
+            }
+            auto offset_str = blob_name.substr(pos);
+            auto offset = std::stol(offset_str);
+            offset_map.emplace(blob_name, offset * kPageSize);
           }
           auto trait = hermes::api::FileMappingTrait(filename, offset_map,
                                                      nullptr, NULL, NULL);
           file_vbucket.Attach(&trait, ctx);
           file_vbucket.Delete(ctx);
+          /**
+           * This is needed as destroy of bucket on line 565 doesn't delete the
+           * blobs.
+           */
```

Comment:
If the blobs aren't deleted that might be a bug. Is there a simple command that lets me reproduce this?

Reply:
I see what's happening. The `Close` on 565 marks the bucket as invalid, and `Destroy` does nothing on an invalid bucket. The way it's supposed to work is that you `Close` a bucket if you want to keep the data around and reopen it later, but you `Destroy` it if you want to delete the blobs and all the bucket's metadata. Removing the `Close` should give you the desired behavior here, as long as this bucket isn't opened by multiple processes.

Reply:
If i do not close then destroy fails saying bucket reference is not 0.

Reply:
What happens is when I call destroy for the first time. 
I0205 16:35:39.246978 46607 metadata_storage_stb_ds.cc:477] Cannot destroy bucket /tmp/test.dat_ext. It's refcount is 2
However, if I close it works.

Reply:
Trace:
```
I0205 16:35:39.243497 46607 stdio.cc:406] Intercept fopen for filename: /tmp/test.dat_ext and mode: w is tracked.
I0205 16:35:39.244196 46607 stdio.cc:44] Open file for filename /tmp/test.dat_ext in mode w
I0205 16:35:39.244225 46607 stdio.cc:52] File not opened before by adapter
I0205 16:35:39.244251 46607 metadata_management.cc:273] Creating Bucket '/tmp/test.dat_ext'
I0205 16:35:39.244275 46607 metadata_manager.cc:11] Create metadata for file handler.
/home/hdevarajan/projects/hermes/adapter/test/stdio/stdio_adapter_test.cpp:211: passed: is_same for: true
/home/hdevarajan/projects/hermes/adapter/test/stdio/stdio_adapter_basic_test.cpp:82: passed: test::fh_orig != nullptr for: 0x0000556297fb85f0 != nullptr
I0205 16:35:39.245244 46607 stdio.cc:587] Intercept fwrite.
I0205 16:35:39.245261 46607 stdio.cc:130] Write called for filename: /tmp/test.dat_ext on offset: 0 and size: 65536
I0205 16:35:39.245277 46607 balanced_mapper.cc:11] Mapping File with offset:0 and size:65536.
I0205 16:35:39.245297 46607 stdio.cc:141] Mapping for read has 1 mapping.
I0205 16:35:39.245329 46607 stdio.cc:153] Create or Overwrite blob 2169613#0 of size:65536.
I0205 16:35:39.246006 46607 bucket.h:155] Attaching blob '2169613#0' to Bucket '/tmp/test.dat_ext'
I0205 16:35:39.246145 46607 metadata_manager.cc:17] Update metadata for file handler.
I0205 16:35:39.246527 46607 stdio.cc:522] Intercept fclose.
I0205 16:35:39.246546 46607 stdio.cc:526] File handler is opened by adapter.
I0205 16:35:39.246562 46607 metadata_manager.cc:52] Delete metadata for file handler.
I0205 16:35:39.246577 46607 stdio.cc:533] Adapter flushes 1 blobs to filename:/tmp/test.dat_ext.
I0205 16:35:39.246596 46607 vbucket.h:39] Create VBucket /tmp/test.dat_ext
I0205 16:35:39.246634 46607 metadata_management.cc:321] Opening VBucket '/tmp/test.dat_ext'
I0205 16:35:39.246665 46607 vbucket.cc:19] Linking blob 2169613#0 in bucket /tmp/test.dat_ext to VBucket /tmp/test.dat_ext
I0205 16:35:39.246703 46607 vbucket.cc:113] Attaching trait to VBucket /tmp/test.dat_ext
I0205 16:35:39.246729 46607 vbucket.cc:193] Deleting VBucket /tmp/test.dat_ext
I0205 16:35:39.246950 46607 bucket.cc:228] Destroying bucket '/tmp/test.dat_ext'
I0205 16:35:39.246978 46607 metadata_storage_stb_ds.cc:477] Cannot destroy bucket /tmp/test.dat_ext. It's refcount is 2
```

Reply:
Comment out the close and the delete of blob and then u can run
```
build/bin/hermes_stdio_adapter_test SingleWrite
```
It will work but u will see the log
```
I0205 16:52:00.818601 49793 metadata_storage_stb_ds.cc:477] Cannot destroy bucket /tmp/test.dat_ext. It's refcount is 2
```

Eventually, this has side effects on the rest of the tests as blob still exists.